src/pages/version/TaskDuration.tsx

```diff
@@ -0,0 +1,152 @@
+import { useEffect } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { useParams, useLocation } from "react-router-dom";
+import { useVersionAnalytics } from "analytics";
+import { PageSizeSelector } from "components/PageSizeSelector";
+import { Pagination } from "components/Pagination";
+import { ResultCountLabel } from "components/ResultCountLabel";
+import { TableControlOuterRow, TableControlInnerRow } from "components/styles";
+import { pollInterval } from "constants/index";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  PatchTaskDurationsQuery,
+  PatchTaskDurationsQueryVariables,
+} from "gql/generated/types";
+import { GET_PATCH_TASK_DURATIONS } from "gql/queries";
+import { usePolling } from "hooks";
+import { useUpdateURLQueryParams } from "hooks/useUpdateURLQueryParams";
+import { PatchTasksQueryParams } from "types/task";
+import { queryString, url } from "utils";
+import { TaskDurationTable } from "./taskDuration/TaskDurationTable";
+
+const { parseQueryString, parseSortString, getString } = queryString;
+
+const { getPageFromSearch, getLimitFromSearch } = url;
+interface Props {
+  taskCount: number;
+}
+
+export const TaskDuration: React.VFC<Props> = ({ taskCount }) => {
+  const { id: versionId } = useParams<{ id: string }>();
+
+  const { search } = useLocation();
+  const { sendEvent } = useVersionAnalytics(versionId);
+  const dispatchToast = useToastContext();
+
+  const updateQueryParams = useUpdateURLQueryParams();
+  const queryVariables = getQueryVariables(search, versionId);
+  const noQueryVariables = !search.length;
+
+  const { limit, page } = queryVariables;
+
+  const { sorts } = parseQueryString(search);
+  const allSorts = parseSortString(sorts);
+  const defaultSortMethod = "STATUS:ASC;BASE_STATUS:DESC;DURATION:DESC";
+
+  useEffect(() => {
+    if (noQueryVariables) {
+      updateQueryParams({
+        sorts: defaultSortMethod,
+      });
+    }
+  }, [search]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const { data, startPolling, stopPolling } = useQuery<
+    PatchTaskDurationsQuery,
+    PatchTaskDurationsQueryVariables
+  >(GET_PATCH_TASK_DURATIONS, {
+    variables: queryVariables,
+    skip: noQueryVariables,
+    pollInterval,
+    fetchPolicy: "network-only",
```

Comment:
Why the network-only fetch policy?

Reply:
Comes from this discussion above https://github.com/evergreen-ci/spruce/pull/1211#discussion_r849642703